picard/ui/coverartbox.py

```diff
@@ -545,6 +550,43 @@ def load_remote_image(self, url, data):
             log.warning("Can't load image: %s", e)
             return
 
+    def delete_cover_art(self):
+        if not self.item or not self.item.metadata.images:
+            return
+
+        cover_art_list = [image.source or _("Unnamed Cover Art") for image in self.item.metadata.images]
+
+        selected_item, ok_pressed = QtWidgets.QInputDialog.getItem(self, _("Delete Cover Art"),
+                                                                    _("Select the cover art image to delete:"),
+                                                                    cover_art_list, 0, False)
+        if ok_pressed:
+            selected_image_index = cover_art_list.index(selected_item)
+            selected_image = self.item.metadata.images[selected_image_index]
+            if not isinstance(self.item, File):
+                self.delete_cover_art_for_item(self.item, selected_image)
+            for file in self.get_files_to_process():
+                self.delete_cover_art_for_item(file, selected_image)
+                self.update()
+            self.item.update()
+
+    def get_files_to_process(self):
+        if isinstance(self.item, (Album, FileListItem)):
+            return self.item.iterfiles()
+        elif isinstance(self.item, File):
+            return [self.item]
+
+    def delete_cover_art_for_item(self, item, selected_image):
+        metadata = item.metadata
+        if not metadata or not metadata.images:
+            return
+
+        try:
+            image_delete(item, selected_image)
+            self.set_item(item)
+            self.cover_art.set_metadata(metadata)
+        except ValueError as e:
+            QtWidgets.QMessageBox.warning(self, _("Error"), str(e))
```

Comment:
This is a warning or an error?

You should rather log an error, and don't display a dialog, because this is called in a loop ...